### PR TITLE
feat: Add doc code snippets and readme generation

### PR DESCRIPTION
**Pull request by nand4011**
Add code snippets for the public dev docs

Add a readme template and update the actions to generate and verify the readme.

Update the examples to gomomento 0.4.0 and make a few fixes to let them build.

Change :hit to :ok in the CacheClient.get() response. This is not yet reflected in the examples/snippets. They will need to be updated after the next release.

### CODE REVIEW

File: .github/workflows/on-pull-request.yml [hunk 1]
```diff
@@ -15,6 +15,15 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
+      - name: Verify README generation
+        uses: momentohq/standards-and-practices/github-actions/oss-readme-template@gh-actions-v2
+        with:
+          project_status: official
+          project_stability: experimental
```
Comment (cprice404) [inline .github/workflows/on-pull-request.yml:22]: let's put this as beta

File: .github/workflows/on-push-to-main-branch.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - name: Setup repo
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.MOMENTO_MACHINE_USER_GITHUB_TOKEN }}
+
+      - name: Generate README
+        uses: momentohq/standards-and-practices/github-actions/generate-and-commit-oss-readme@gh-actions-v2
+        with:
+          project_status: official
+          project_stability: experimental
```
Comment (cprice404) [inline .github/workflows/on-push-to-main-branch.yml:20]: same here

File: README.template.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{{ ossHeader }}
+
+## Packages
+
+The Elixir SDK is available on Hex:
+
+```elixir
+defp deps do
+    [
+      {:gomomento, "~> 0.4.0"}
```
Comment (cprice404) [inline README.template.md:10]: if you put the version number here you will have to figure out how to update it automatically or remember to do it every time we do a release. I would probably just include a link to the hex page in your text content.

**Review by cprice404 — COMMENTED** [review]
looks great, just a couple of minor suggestions

File: README.template.md (hunk 3 above)
Comment (nand4011) [reply README.template.md:10]: Good call.

**Review by cprice404 — APPROVED** [review]

**Pull request merged**